src/Compilers/CSharp/Portable/Symbols/Source/LambdaSymbol.cs

```diff
@@ -255,7 +256,20 @@ public override ImmutableArray<Location> Locations
         {
             get
             {
-                return ImmutableArray.Create<Location>(_syntax.Location);
```

Comment:
It looks like some refactorings are affected (`MakeMethodSynchronousTests.TestSimpleLambda`), which supports your point.
Let me see if I can find an alternative approach.